speech endpoints demo (#3765)
openvinotoolkit/model_server: demos/audio/README.md

# How to serve audio models via OpenAI API {#ovms_demos_audio}

This demo shows how to deploy audio in the OpenVINO Model Server.
Speech generation and speech recognition use cases are exposed via OpenAI API `audio/speech` end `audio/transcriptions` endpoint.

## Prerequisites

**OVMS version 2025.4** This demo require version 2025.4 or nightly release.

**Model preparation**: Python 3.12 or higher with pip

**Model Server deployment**: Installed Docker Engine or OVMS binary package according to the [baremetal deployment guide](../../../docs/deploying_server_baremetal.md)

**Client**: curl or Python for using OpenAI client package

## Speech generation
----
### Model preparation

> Only supported model for speech generation use case is microsoft/speecht5_tts which is needs to be converted to IR format before using in OVMS.

Reviewer: ```suggestion
Supported models should use the topology of [microsoft/speecht5_tts](https://huggingface.co/microsoft/speecht5_tts)which is needs to be converted to IR format before using in OVMS.
```

**(Optional) Client**: git and Python for using OpenAI client package

## Speech generation
### Model preparation
Only supported model for speech generation use case is microsoft/speecht5_tts which is outside OpenVINO organization and needs to be converted to IR format before using in OVMS.

Specific OVMS pull mode example for models outside of OpenVINO organization is described in section `## Pulling models outside of OpenVINO organization` in the [Ovms pull mode](https://github.com/openvinotoolkit/model_server/blob/main/docs/pull_hf_models.md)

Or you can use the python export_model.py script described below.

Here, the original TTS model will be converted to IR format and optionally quantized.
That ensures faster initialization time, better performance and lower memory consumption.
Execution parameters will be defined inside the `graph.pbtxt` file.

Download export script, install it's dependencies and create directory for the models:
```console
curl https://raw.githubusercontent.com/openvinotoolkit/model_server/refs/heads/releases/2025/3/demos/common/export_models/export_model.py -o export_model.py

> pip3 install -r https://raw.githubusercontent.com/openvinotoolkit/model_server/refs/heads/releases/2025/3/demos/common/export_models/requirements.txt

Reviewer 2: use from main?
export_model.py: error: argument task: invalid choice: 'text_to_speech' (choose from 'text_generation', 'embeddings', 'embeddings_ov', 'rerank', 'rerank_ov', 'image_generation')
